tests/Model/Transaction.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+declare(strict_types=1);
+
+namespace atk4\data\tests\Model;
+
+use atk4\data\Model\Union;
+
+class Transaction extends Union
+{
+    public $nestedInvoice;
+    public $nestedPayment;
+
+    public $subtractInvoice;
+
+    public function init(): void
+    {
+        parent::init();
+
+        // first lets define nested models
+        $this->nestedInvoice = $this->addNestedModel(new Invoice(), $this->subtractInvoice ? ['amount' => '-[]'] : []);
```

Comment:
I strongly belive that nesting/wrapping should be standard property of Model, ie. we should either allow `table` to be a string or child Model.

One way to solve this is to define named models with registry which will produce registration key/string which can be used as a table.

Reply:
`$model->table = $another_model` would be nice to support. Also dsql already should support that.
But I guess that's not a functionality of this specific PR.

Reply:
but what is the purpose of this PR then, if we should introduce the support in completely different way? ;-)